### DISCUSSION TITLE
Adding Rule Discovery [Indirect application of Matrix Profile]

### DISCUSSION THREAD

**Discussion by NimaSarajpoor** [TDAmeritrade/stumpy#388] [Ideas]
In Time Series, sometimes we are interested in knowing if a particular sequence will occur after observing another sequence. (In other words: Rule Discovery)

**Papers:**

[Discovery of Meaningful Rules in Time Series [2015]](https://dl.acm.org/doi/10.1145/2783258.2783306)

[Motif-based Rule Discovery for Predicting Real-valued Time Series [2017]
](https://arxiv.org/abs/1709.04763)

They are not directly related to Matrix Profile, but I think it can be added to the package as Matrix Profile can act as the input for such function. (I will go through the papers if time allows me to dig deeper into this matter.)

**Comment by seanlaw** [comment]
@Ninimama While I appreciate your enthusiasm. Can you please provide direct links to those papers in case others would like to contribute too? Thanks in advance!

Additionally, I am open to your suggestion but I will caution that `stumpy` does not aim to be "everything for everyone" and the scope should be kept small where possible. Otherwise, it becomes very hard to maintain in the longer term. At its core `stumpy` efficiently computes the matrix profile and it should serve as the foundation for other packages.

**Comment by NimaSarajpoor** [comment]
@seanlaw 
I totally understand. I will provide the links soon but feel free to close the issue if you think it is out of the scope (and You are right). 

Just a suggestion (not sure if it makes sense or not):

There is scikit-learn-extra package that can be considered as the supplementary package for scikit-learn and extends its modules. You might be interested in creating a similar thing later if you have the time. I think I saw rule discovery problem as one of those 100 problems of time series mentioned in another issue.

**Comment by NimaSarajpoor** [comment]
@seanlaw 

> Can you please provide direct links to those papers in case others would like to contribute too?

I revised the issue and provided the link.

**Comment by seanlaw** [comment]
> There is scikit-learn-extra package that can be considered as the supplementary package for scikit-learn and extends its modules. 

Exactly! While I may not have the bandwidth to create the extensions, hopefully, others will build on top of our STUMPY foundation. This has always been our goal and also why we've ensured 100% unit test coverage since day one.

> I revised the issue and provided the link.

Thank you

**Comment by seanlaw** [comment]
> In Time Series, sometimes we are interested in knowing if a particular sequence will occur after observing another sequence. (In other words: Rule Discovery)
> 
> **Papers:**
> 
> [Discovery of Meaningful Rules in Time Series [2015]](https://dl.acm.org/doi/10.1145/2783258.2783306)
> 
> [Motif-based Rule Discovery for Predicting Real-valued Time Series [2017] ](https://arxiv.org/abs/1709.04763)
> 
> They are not directly related to Matrix Profile, but I think it can be added to the package as Matrix Profile can act as the input for such function. (I will go through the papers if time allows me to dig deeper into this matter.)

@Ninimama I moved some of the Github Issues over into this Discussion forum. Once we've (collectively) decided that it is worth pursuing then we can create a new focused issue in the Github Issues section. This way:

1. We can have much longer freeform discussions here where we can evolve the idea into something
2. We keep the issues as a backlog for things that need to be completed
3. Github Issues are more about specific code around well formed ideas (especially published papers) while Discussions are still being formed and may or may not result in any code 

Please let me know if that makes sense

**Reply by NimaSarajpoor** [reply]
Totally makes sense for keeping this repo and the library organized
